server/src/main/java/org/elasticsearch/common/component/AbstractLifecycleComponent.java

```diff
@@ -34,6 +34,13 @@
 
     private final List<LifecycleListener> listeners = new CopyOnWriteArrayList<>();
 
+    protected AbstractLifecycleComponent() {}
+
+    /**
+     * @deprecated the settings parameters are not used, therefore the use of this constructor is deprecated.
```

Comment:
Can you please indicate which constructor should be called instead (for completeness)?